### ISSUE TITLE
Issue with UUPMediaConverter on linux

### ISSUE THREAD

**Issue by buddyjojo** [gus33000/UUPMediaCreator#129]
When I try to use UUPMediaConverter (v0.3.1.1 linux-x64) on linux (debian sid) i get this error:
`
System.DllNotFoundException: Unable to load shared library 'WimgApi.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libWimgApi.dll: cannot open shared object file: No such file or directory`

Here's the full log:

<details>
<summary>Log</summary>
<br>

```
sudo ./UUPMediaConverter  -u 10.0.22000.318.co_release_svc_prod1.211102-1556_arm64fre_86abe5724c69/ -i win.iso -l en-US -e Core
[01:52:32][Information] UUPMediaConverter 0.3.1.1 - Converts an UUP file set to an usable ISO file
[01:52:32][Information] Copyright (c) Gustave Monce and Contributors
[01:52:32][Information] https://github.com/gus33000/UUPMediaCreator

[01:52:32][Information] This program comes with ABSOLUTELY NO WARRANTY.
[01:52:32][Information] This is free software, and you are welcome to redistribute it under certain conditions.

[01:52:32][Information] This software contains work derived from libmspack licensed under the LGPL-2.1 license.
[01:52:32][Information] (C) 2003-2004 Stuart Caie.
[01:52:32][Information] (C) 2011 Ali Scissons.

[01:52:32][  Warning  ] WARNING: For successful ISO creation, please install genisoimage
[01:52:32][  Warning  ] WARNING: This tool does NOT currently integrate updates into the finished media file. Any UUP set with updates (KBXXXXX).MSU/.CAB will not have the update integrated.
[01:52:32][  Warning  ] WARNING: This tool is NOT currently running under Windows as administrator. The resulting image will be less clean/proper compared to Microsoft original.
[01:52:37][Information] [ReadingMetadata] Acquiring Composition Databases
[01:52:37][Information] [ReadingMetadata] Acquiring Base Editions
[01:52:37][Information] [ReadingMetadata] Adding Base Editions to the conversion plan
[01:52:37][Information] [ReadingMetadata] Building Targets
[01:52:37][Information] [ReadingMetadata] One edition Composition Database failed file validation, below is highlighted the files that could not be found in the UUP path. This means that you will not get all possible editions.
[01:52:37][Information] [ReadingMetadata] Missing: UUP/Desktop/editionPackages/neutral/Microsoft-Windows-EditionSpecific-Professional-Package.ESD
[01:52:37][Information] [ReadingMetadata] Missing: UUP/Desktop/editionPackages/neutral/Microsoft-Windows-EditionPack-Professional-Package.ESD
[01:52:37][Information] [ReadingMetadata] Missing: UUP/Desktop/editionPackages/neutral/Microsoft-Windows-EditionPack-Professional-WOW64-Package.ESD
[01:52:37][Information] [ReadingMetadata] Missing: UUP/Desktop/editionPackages/neutral/Microsoft-Windows-EditionPack-Professional-arm64arm-Package.ESD
[01:52:37][Information] [ReadingMetadata] Missing: UUP/Desktop/Apps/Microsoft.ModernApps.Client.professional.esd
[01:52:37][Information] [ReadingMetadata] Missing: FeaturesOnDemand/neutral/cabs/Microsoft-Windows-Printing-PMCPPC-FoD-Package~31bf3856ad364e35~arm64~~.cab
[01:52:37][Information] [ReadingMetadata] Missing: MetadataESD/professional_en-us.esd
[01:52:37][Information] [ReadingMetadata] -> Name: core, Availability: Canonical
[01:52:37][Information] [ReadingMetadata] Enumerating files
[01:52:37][Information] [ReadingMetadata] Looking up Composition Database in order to find a Base ESD image appropriate for building windows setup files.
[01:52:37][Information] [CreatingWindowsInstaller] Applying file structure (0%)
[01:52:37][Information] [CreatingWindowsInstaller] Applying file structure (100%)
[01:52:37][Information] [CreatingWindowsInstaller][0%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][1%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][2%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][3%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][4%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][5%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][6%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][7%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][8%] Applying core_en-us.esd - Index 1
[01:52:37][Information] [CreatingWindowsInstaller][9%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][10%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][11%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][12%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][13%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][14%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][15%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][16%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][17%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][18%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][19%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][20%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][21%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][22%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][23%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][24%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][25%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][26%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][27%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][29%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][31%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][33%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][34%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][36%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][37%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][38%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][40%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][41%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][42%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][43%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][44%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][46%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][47%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][48%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][49%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][50%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][51%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][54%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][55%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][56%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][57%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][58%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][59%] Applying core_en-us.esd - Index 1
[01:52:38][Information] [CreatingWindowsInstaller][63%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][66%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][67%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][68%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][70%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][71%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][72%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][73%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][75%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][76%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][77%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][78%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][79%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][80%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][81%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][86%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][87%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][88%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][90%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][91%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][93%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][94%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][95%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][96%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][97%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][98%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][99%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller][100%] Applying core_en-us.esd - Index 1
[01:52:39][Information] [CreatingWindowsInstaller] Applying metadata (100%)
[01:52:39][Information] [CreatingWindowsInstaller] Extracting XML Lite
[01:52:39][   Error   ] An error occured!
[01:52:39][   Error   ] System.DllNotFoundException: Unable to load shared library 'WimgApi.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libWimgApi.dll: cannot open shared object file: No such file or directory
   at Microsoft.Wim.WimgApi.NativeMethods.WIMCreateFile(String pszWimPath, UInt32 dwDesiredAccess, UInt32 dwCreationDisposition, UInt32 dwFlagsAndAttributes, UInt32 dwCompressionType, WimCreationResult& pdwCreationResult)
   at Microsoft.Wim.WimgApi.CreateFile(String path, WimFileAccess desiredAccess, WimCreationDisposition creationDisposition, WimCreateFileOptions options, WimCompressionType compressionType) in D:\a\UUPMediaCreator\UUPMediaCreator\thirdparty\Microsoft.Wim.NET\WimgApi.CreateFile.cs:line 131
   at Imaging.WimgApiImaging.ExtractFileFromImage(String wimFile, Int32 imageIndex, String fileToExtract, String destination) in D:\a\UUPMediaCreator\UUPMediaCreator\src\Imaging.NET\WimgApiImaging.cs:line 326
   at Imaging.WimImaging.ExtractFileFromImage(String wimFile, Int32 imageIndex, String fileToExtract, String destination) in D:\a\UUPMediaCreator\UUPMediaCreator\src\Imaging.NET\WIMImaging.cs:line 44
   at MediaCreationLib.Installer.WindowsInstallerBuilder.CreateSetupMediaRoot(String BaseESD, String OutputPath, ProgressCallback progressCallback) in D:\a\UUPMediaCreator\UUPMediaCreator\src\MediaCreationLib.NET\Installer\WindowsInstallerBuilder.cs:line 644
   at MediaCreationLib.Installer.WindowsInstallerBuilder.BuildSetupMedia(String BaseESD, String OutputWinREPath, String MediaPath, WimCompressionType compressionType, Boolean RunsAsAdministrator, String LanguageCode, TempManager tempManager, ProgressCallback progressCallback) in D:\a\UUPMediaCreator\UUPMediaCreator\src\MediaCreationLib.NET\Installer\WindowsInstallerBuilder.cs:line 52
   at MediaCreationLib.Installer.SetupMediaCreator.CreateSetupMedia(String UUPPath, String LanguageCode, String OutputMediaPath, String OutputWindowsREPath, CompressionType CompressionType, IEnumerable`1 CompositionDatabases, TempManager tempManager, ProgressCallback progressCallback) in D:\a\UUPMediaCreator\UUPMediaCreator\src\MediaCreationLib.NET\Installer\SetupMediaCreator.cs:line 71
   at MediaCreationLib.MediaCreator.CreateISOMedia(String ISOPath, String UUPPath, String Edition, String LanguageCode, Boolean IntegrateUpdates, CompressionType CompressionType, ProgressCallback progressCallback, String Temp) in D:\a\UUPMediaCreator\UUPMediaCreator\src\MediaCreationLib.NET\MediaCreator.cs:line 414

```
</details>

I have genisoimage installed.

Am I doing something wrong?

**Comment by gus33000** [comment]
Hello this is meant to be fixed for a while already, (sorry just noticed this issue). Is it ok on your side with newer versions?

**Comment by buddyjojo** [comment]
I'll try when I've got the chance

**Comment by mb3** [comment]
I saw the same error (`Unable to load shared library 'WimgApi.dll'`) with v3.1.4.0 on macOS recently.

<details><summary>Log</summary>

```
./UUPMediaConverter -u 10.0.22621.819.ni_release_svc_prod1.221103-1146_arm64fre_4bae0a813495 -i ./Win11_22H2_arm64.iso -l en-US
[00:21:23][Information] UUPMediaConverter 3.1.4.0 - Converts an UUP file set to an usable ISO file
[00:21:23][Information] Copyright (c) Gustave Monce and Contributors
[00:21:23][Information] https://github.com/gus33000/UUPMediaCreator

[00:21:23][Information] This program comes with ABSOLUTELY NO WARRANTY.
[00:21:23][Information] This is free software, and you are welcome to redistribute it under certain conditions.

[00:21:23][Information] This software contains work derived from libmspack licensed under the LGPL-2.1 license.
[00:21:23][Information] (C) 2003-2004 Stuart Caie.
[00:21:23][Information] (C) 2011 Ali Scissons.

[00:21:23][  Warning  ] WARNING: For successful ISO creation, please install cdrtools via brew
[00:21:23][  Warning  ] WARNING: This tool does NOT currently integrate updates into the finished media file. Any UUP set with updates (KBXXXXX).MSU/.CAB will not have the update integrated.
[00:21:23][  Warning  ] WARNING: This tool is NOT currently running under Windows as administrator. The resulting image will be less clean/proper compared to Microsoft original.
[00:21:23][  Warning  ] WARNING: You are attempting to create an ISO media with potentially all editions available. Due to the tool not running under Windows as administrator, this request might not be fullfilled.
[00:21:30][Information] [ReadingMetadata] Acquiring Composition Databases
[00:21:30][Information] [ReadingMetadata] Acquiring Base Editions
[00:21:30][Information] [ReadingMetadata] Adding Base Editions to the conversion plan
[00:21:30][Information] [ReadingMetadata] Building Targets
[00:21:30][Information] [ReadingMetadata] One edition Composition Database failed file validation, below is highlighted the files that could not be found in the UUP path. This means that you will not get all possible editions.
[00:21:30][Information] [ReadingMetadata] Missing: UUP/Desktop/editionPackages/neutral/Microsoft-Windows-EditionSpecific-Core-Package.ESD
[00:21:30][Information] [ReadingMetadata] Missing: UUP/Desktop/editionPackages/neutral/Microsoft-Windows-EditionPack-Core-Package.ESD
[00:21:30][Information] [ReadingMetadata] Missing: UUP/Desktop/editionPackages/neutral/Microsoft-Windows-EditionPack-Core-WOW64-Package.ESD
[00:21:30][Information] [ReadingMetadata] Missing: UUP/Desktop/editionPackages/neutral/Microsoft-Windows-EditionPack-Core-arm64arm-Package.ESD
[00:21:30][Information] [ReadingMetadata] Missing: MetadataESD/core_en-us.esd
[00:21:30][Information] [ReadingMetadata] -> Name: professional, Availability: Canonical
[00:21:30][Information] [ReadingMetadata] Enumerating files
[00:21:30][Information] [ReadingMetadata] Looking up Composition Database in order to find a Base ESD image appropriate for building windows setup files.
[00:21:30][Information] [CreatingWindowsInstaller] Applying file structure (0%)
[00:21:30][Information] [CreatingWindowsInstaller] Applying file structure (100%)
[00:21:30][Information] [CreatingWindowsInstaller][0%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][1%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][2%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][3%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][4%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][5%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][6%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][7%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][8%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][9%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][10%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][11%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][12%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][13%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][14%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][15%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][16%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][17%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][18%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][19%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][20%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][21%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][22%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][23%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][24%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][25%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][26%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][27%] Applying professional_en-us.esd - Index 1
[00:21:30][Information] [CreatingWindowsInstaller][28%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][29%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][32%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][34%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][35%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][38%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][39%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][40%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][41%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][42%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][43%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][44%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][45%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][47%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][48%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][49%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][50%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][51%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][52%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][54%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][56%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][57%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][58%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][59%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][60%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][61%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][66%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][67%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][70%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][71%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][72%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][73%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][74%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][75%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][76%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][77%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][78%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][79%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][80%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][82%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][83%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][84%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][86%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][87%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][88%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][89%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][91%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][92%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][94%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][95%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][96%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][97%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][98%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][99%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller][100%] Applying professional_en-us.esd - Index 1
[00:21:31][Information] [CreatingWindowsInstaller] Applying metadata (100%)
[00:21:31][Information] [CreatingWindowsInstaller] Extracting XML Lite
[00:21:31][   Error   ] An error occured!
[00:21:31][   Error   ] System.DllNotFoundException: Unable to load shared library 'WimgApi.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(libWimgApi.dll, 0x0001): tried: 'libWimgApi.dll' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibWimgApi.dll' (no such file), '/usr/lib/libWimgApi.dll' (no such file, not in dyld cache), 'libWimgApi.dll' (no such file), '/usr/local/lib/libWimgApi.dll' (no such file), '/usr/lib/libWimgApi.dll' (no such file, not in dyld cache)
   at Microsoft.Wim.WimgApi.NativeMethods.WIMCreateFile(String pszWimPath, UInt32 dwDesiredAccess, UInt32 dwCreationDisposition, UInt32 dwFlagsAndAttributes, UInt32 dwCompressionType, WimCreationResult& pdwCreationResult)
   at Microsoft.Wim.WimgApi.CreateFile(String path, WimFileAccess desiredAccess, WimCreationDisposition creationDisposition, WimCreateFileOptions options, WimCompressionType compressionType) in D:\a\UUPMediaCreator\UUPMediaCreator\thirdparty\Microsoft.Wim\WimgApi.CreateFile.cs:line 131
   at Imaging.NET.WimgApiImaging.ExtractFileFromImage(String wimFile, Int32 imageIndex, String fileToExtract, String destination) in D:\a\UUPMediaCreator\UUPMediaCreator\src\Imaging.NET\WimgApiImaging.cs:line 333
   at Imaging.NET.WimImaging.ExtractFileFromImage(String wimFile, Int32 imageIndex, String fileToExtract, String destination) in D:\a\UUPMediaCreator\UUPMediaCreator\src\Imaging.NET\WIMImaging.cs:line 44
   at MediaCreationLib.NET.Installer.WindowsInstallerBuilder.CreateSetupMediaRoot(String BaseESD, String OutputPath, ProgressCallback progressCallback) in D:\a\UUPMediaCreator\UUPMediaCreator\src\MediaCreationLib.NET\Installer\WindowsInstallerBuilder.cs:line 645
   at MediaCreationLib.NET.Installer.WindowsInstallerBuilder.BuildSetupMedia(String BaseESD, String OutputWinREPath, String MediaPath, WimCompressionType compressionType, Boolean RunsAsAdministrator, String LanguageCode, TempManager tempManager, ProgressCallback progressCallback) in D:\a\UUPMediaCreator\UUPMediaCreator\src\MediaCreationLib.NET\Installer\WindowsInstallerBuilder.cs:line 53
   at MediaCreationLib.NET.Installer.SetupMediaCreator.CreateSetupMedia(String UUPPath, String LanguageCode, String OutputMediaPath, String OutputWindowsREPath, CompressionType CompressionType, IEnumerable`1 CompositionDatabases, TempManager tempManager, ProgressCallback progressCallback) in D:\a\UUPMediaCreator\UUPMediaCreator\src\MediaCreationLib.NET\Installer\SetupMediaCreator.cs:line 70
   at MediaCreationLib.NET.MediaCreator.CreateISOMedia(String ISOPath, String UUPPath, String Edition, String LanguageCode, Boolean IntegrateUpdates, CompressionType CompressionType, ProgressCallback progressCallback, String Temp) in D:\a\UUPMediaCreator\UUPMediaCreator\src\MediaCreationLib.NET\MediaCreator.cs:line 413
```

</details>

(As recommended, I have `cdrtools` installed.)

**Comment by gus33000** [comment]
Hey, this issue just got fixed with v3.1.5.0, sorry for the delay!

Please give it a try and tell us if you hit any issue.

https://github.com/gus33000/UUPMediaCreator/releases/tag/v3.1.5.0
